server/bin/basket-proxy-server.js

```diff
@@ -0,0 +1,180 @@
+#!/usr/bin/env node
+
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+'use strict';
+
+var url = require('url');
+var mozlog = require('mozlog');
+var request = require('request');
+
+var config = require('../lib/configuration');
+mozlog.config(config.get('logging'));
+
+var logger = require('mozlog')('server.basketproxy');
+
+var express = require('express');
+var bodyParser = require('body-parser');
+var cors = require('cors');
+
+var CORS_ORIGIN = config.get('public_url');
+var API_KEY = config.get('basket.api_key');
+var API_URL = config.get('basket.api_url');
+var VERIFY_URL = config.get('oauth_url') + '/v1/verify';
+
+
+// Error codes are defined in:
+// https://github.com/mozilla/basket-client/blob/master/basket/errors.py
+var BASKET_ERRORS = {
+  NETWORK_FAILURE: 1,
+  INVALID_EMAIL: 2,
+  UNKNOWN_EMAIL: 3,
+  UNKNOWN_TOKEN: 4,
+  USAGE_ERROR: 5,
+  EMAIL_PROVIDER_AUTH_FAILURE: 6,
+  AUTH_ERROR: 7,
+  SSL_REQUIRED: 8,
+  INVALID_NEWSLETTER: 9,
+  INVALID_LANGUAGE: 10,
+  EMAIL_NOT_CHANGED: 11,
+  CHANGE_REQUEST_NOT_FOUND: 12,
+
+  // If you get this, report it as a bug so we can add a more specific
+  // error code.
+  UNKNOWN_ERROR: 99
+};
+
+function errorResponse(desc, code) {
+  // Format from
+  // https://basket.readthedocs.org/en/latest/newsletter_api.html
+  return {
+    status: 'error',
+    desc: String(desc),
+    code: code || BASKET_ERRORS.UNKNOWN_ERROR
+  };
+}
+
+// Verify an OAuth token and return the associated credentials
+function verifyOAuthToken() {
+  return function (req, res, next) {
+    var authHeader = req.headers && req.headers.authorization;
+
+    if (! authHeader) {
+      logger.error('auth.missing-authorization-header');
+      res.status(400).json(errorResponse('missing authorization header', BASKET_ERRORS.USAGE_ERROR));
+      return;
+    }
+
+    var token = authHeader.replace(/^Bearer /, '');
```

Comment:
IMHO it would be better to explicitly check for "Bearer" and error out if there's some other weirdy auth type in the header, e.g.:

```
var authz = authHeader..split(/\s+/);
if (authz.length !== 2 || authz[0].toLowerCase() !== "bearer") {
  // error out
}
```